### ISSUE TITLE
Trouble with quality of results using pretrained model

### ISSUE THREAD

**Issue by dwallener** [csqiangwen/DeepFillv2_Pytorch#8]
Used the pretrained model against a test image of my own, and the results are not good at all. Curious if there are any thoughts on what might be causing this. Everything seems to run correctly.

![03-with-cutout](https://user-images.githubusercontent.com/5768048/122691317-c4889980-d1e3-11eb-8c7f-35f06f2c5451.png)
![03-mask](https://user-images.githubusercontent.com/5768048/122691320-c7838a00-d1e3-11eb-8f04-8a1080b6def2.png)
![5_second_out (1)](https://user-images.githubusercontent.com/5768048/122691344-f26dde00-d1e3-11eb-8916-d6312af1e552.jpg)

**Comment by csqiangwen** [comment]
It is a little weird. This case should be easily handled by this inpainting algorithm.

Have you tried another image?

**Comment by wmmin** [comment]
![Uploading image.png…]()
I also hava a same problem,use your pretrained model,the result was bad.And I used your code to traing celeba_hq datasets,the results also wsa bad.Why is it happen?

**Comment by wmnj** [comment]
me too

**Comment by Tiuuuuu** [comment]
me too

**Comment by AlonzoLeeeooo** [comment]
> Excuse me. My training samples are also poor in quality. (trained for about 125K iterations) Do I need to train it longer? Or the model itself is mistaken? ![epoch6batch0_gt](https://user-images.githubusercontent.com/68578066/136791744-d899dc18-1490-49b6-8c6c-0e4e16df14be.png) ![epoch6batch0_second_out](https://user-images.githubusercontent.com/68578066/136791755-8419fbf0-9899-4a66-994f-06086ebed5b9.png) P.S. The 1st pic is ground truth, the 2nd pic is the result. (Also it was in 256 resolution but the paper said it can reach 512)

I have tried another PyTorch code implementation of DeepFill-v2 using [mmcv](https://github.com/open-mmlab/mmcv) and it works. You may refer to it if you are interested.

**Comment by poinne** [comment]
> > Excuse me. My training samples are also poor in quality. (trained for about 125K iterations) Do I need to train it longer? Or the model itself is mistaken? ![epoch6batch0_gt](https://user-images.githubusercontent.com/68578066/136791744-d899dc18-1490-49b6-8c6c-0e4e16df14be.png) ![epoch6batch0_second_out](https://user-images.githubusercontent.com/68578066/136791755-8419fbf0-9899-4a66-994f-06086ebed5b9.png) P.S. The 1st pic is ground truth, the 2nd pic is the result. (Also it was in 256 resolution but the paper said it can reach 512)
> 
> I have tried another PyTorch code implementation of DeepFill-v2 using [mmcv](https://github.com/open-mmlab/mmcv) and it works. You may refer to it if you are interested.
How did you train DeepFill using mmcv？

**Comment by AlonzoLeeeooo** [comment]
> > > Excuse me. My training samples are also poor in quality. (trained for about 125K iterations) Do I need to train it longer? Or the model itself is mistaken? ![epoch6batch0_gt](https://user-images.githubusercontent.com/68578066/136791744-d899dc18-1490-49b6-8c6c-0e4e16df14be.png) ![epoch6batch0_second_out](https://user-images.githubusercontent.com/68578066/136791755-8419fbf0-9899-4a66-994f-06086ebed5b9.png) P.S. The 1st pic is ground truth, the 2nd pic is the result. (Also it was in 256 resolution but the paper said it can reach 512)
> > 
> > 
> > I have tried another PyTorch code implementation of DeepFill-v2 using [mmcv](https://github.com/open-mmlab/mmcv) and it works. You may refer to it if you are interested.
> > How did you train DeepFill using mmcv？

125k iterations might not be sufficient to reach the optimal performance of DeepFill-v2. I recommend training DeepFill-v2 with 300k iterations or above.

**Comment by poinne** [comment]
> > > > Excuse me. My training samples are also poor in quality. (trained for about 125K iterations) Do I need to train it longer? Or the model itself is mistaken? ![epoch6batch0_gt](https://user-images.githubusercontent.com/68578066/136791744-d899dc18-1490-49b6-8c6c-0e4e16df14be.png) ![epoch6batch0_second_out](https://user-images.githubusercontent.com/68578066/136791755-8419fbf0-9899-4a66-994f-06086ebed5b9.png) P.S. The 1st pic is ground truth, the 2nd pic is the result. (Also it was in 256 resolution but the paper said it can reach 512)
> > > 
> > > 
> > > I have tried another PyTorch code implementation of DeepFill-v2 using [mmcv](https://github.com/open-mmlab/mmcv) and it works. You may refer to it if you are interested.
> > > How did you train DeepFill using mmcv？
> 
> 125k iterations might not be sufficient to reach the optimal performance of DeepFill-v2. I recommend training DeepFill-v2 with 300k iterations or above.

Nonono, I'm curious about how you implemented DeepFill using mmcv. Did you rewrite the network structure of DeepFill using components from mmcv? If possible, could you provide me with a reference code?

**Comment by AlonzoLeeeooo** [comment]
> > > > > Excuse me. My training samples are also poor in quality. (trained for about 125K iterations) Do I need to train it longer? Or the model itself is mistaken? ![epoch6batch0_gt](https://user-images.githubusercontent.com/68578066/136791744-d899dc18-1490-49b6-8c6c-0e4e16df14be.png) ![epoch6batch0_second_out](https://user-images.githubusercontent.com/68578066/136791755-8419fbf0-9899-4a66-994f-06086ebed5b9.png) P.S. The 1st pic is ground truth, the 2nd pic is the result. (Also it was in 256 resolution but the paper said it can reach 512)
> > > > 
> > > > 
> > > > I have tried another PyTorch code implementation of DeepFill-v2 using [mmcv](https://github.com/open-mmlab/mmcv) and it works. You may refer to it if you are interested.
> > > > How did you train DeepFill using mmcv？
> > 
> > 
> > 125k iterations might not be sufficient to reach the optimal performance of DeepFill-v2. I recommend training DeepFill-v2 with 300k iterations or above.
> 
> Nonono, I'm curious about how you implemented DeepFill using mmcv. Did you rewrite the network structure of DeepFill using components from mmcv? If possible, could you provide me with a reference code?

MMCV has an official code. You can directly use that implementation. Details can be referred to [this website](https://github.com/open-mmlab/mmagic/blob/main/configs/deepfillv2/README.md).

**Comment by poinne** [comment]
> > > > > > Excuse me. My training samples are also poor in quality. (trained for about 125K iterations) Do I need to train it longer? Or the model itself is mistaken? ![epoch6batch0_gt](https://user-images.githubusercontent.com/68578066/136791744-d899dc18-1490-49b6-8c6c-0e4e16df14be.png) ![epoch6batch0_second_out](https://user-images.githubusercontent.com/68578066/136791755-8419fbf0-9899-4a66-994f-06086ebed5b9.png) P.S. The 1st pic is ground truth, the 2nd pic is the result. (Also it was in 256 resolution but the paper said it can reach 512)
> > > > > 
> > > > > 
> > > > > I have tried another PyTorch code implementation of DeepFill-v2 using [mmcv](https://github.com/open-mmlab/mmcv) and it works. You may refer to it if you are interested.
> > > > > How did you train DeepFill using mmcv？
> > > 
> > > 
> > > 125k iterations might not be sufficient to reach the optimal performance of DeepFill-v2. I recommend training DeepFill-v2 with 300k iterations or above.
> > 
> > 
> > Nonono, I'm curious about how you implemented DeepFill using mmcv. Did you rewrite the network structure of DeepFill using components from mmcv? If possible, could you provide me with a reference code?
> 
> MMCV has an official code. You can directly use that implementation. Details can be referred to [this website](https://github.com/open-mmlab/mmagic/blob/main/configs/deepfillv2/README.md).

Sure, thank you very much!
